include/vrv/iomidi.h

```diff
@@ -0,0 +1,75 @@
+/////////////////////////////////////////////////////////////////////////////
+// Name:        iomidi.h
+// Author:      Klaus Rettinghaus
+// Created:     2016
+// Copyright (c) Authors and others. All rights reserved.
+/////////////////////////////////////////////////////////////////////////////
+
+#ifndef __VRV_IOMIDI_H__
+#define __VRV_IOMIDI_H__
+
+#include <sstream>
+
+//----------------------------------------------------------------------------
+
+#include "doc.h"
+#include "io.h"
+
+//----------------------------------------------------------------------------
+
+#ifndef NO_MIDI_SUPPORT
+#include "MidiFile.h"
+#endif /* NO_MIDI_SUPPORT */
+
+//----------------------------------------------------------------------------
+
+class MidiFile;
+
+namespace vrv {
+
+//----------------------------------------------------------------------------
+// MidiOutput
+//----------------------------------------------------------------------------
+
+class MidiOutput : public FileOutputStream {
+public:
+    /** @name Constructors and destructor */
+    ///@{
+    MidiOutput(Doc *doc, std::string filename);
+    virtual ~MidiOutput();
+    ///@}
+
+    /**
+     * The main method for exporting the file to MEI.
+     */
+    virtual bool ExportFile();
+
+    /**
+     * Export the document to a MIDI file.
+     * Run trough all the layer and fill the midi file content.
+     */
+    void ExportMIDI(MidiFile *midiFile);
```

Comment:
I guess the idea is to move the content of Doc::ExportMIDI here. This is good. However, was we should do it to have the MidiFile class inside the MidiOutput class (and not the Toolkit). We could make it aligned with the MeiOutput class, namely to have an ExportFile() method and a GetOutput() method. The second one returns the output as a std::string, which has to be 64based encoded for MIDI. This is already happening in Toolkit::RenderToMidi